Exclude java/nio/channels/AsyncCloseAndInterrupt.java aix-all,linux-all

JasonFengJ9 opened the pull request:
Exclude `java/nio/channels/AsyncCloseAndInterrupt.java` `aix-all,linux-all`

Match Openjdk and other OpenJ9 version excludes.

Related to
*  https://github.com/eclipse-openj9/openj9/issues/21777

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

pshipton commented:
What about 21?

pshipton commented:
Why is it being excluded on linux, does it fail on linux too?

JasonFengJ9 commented:
`java/nio/channels/AsyncCloseAndInterrupt.java` was excluded for `aix-all,linux-all` in all other `ProblemList_openjdk_xxx.txt` files. This is to match OpenJDK and other OpenJ9 versions.

pshipton commented:
But it it's not failing for Semeru on Linux, why would we exclude it there? If it is failing on Linux, the openj9 issue(s) should mention this. We don't need to duplicate what adoptium is doing. 

I see excludes for dead version and for jdk24+, but not for jdk21. Does it not fail there as well?

pshipton commented:
For the excludes jdk24+, they are excluded via https://github.com/adoptium/infrastructure/issues/1173 but this issue is closed saying the tests run successfully and the problematic machines are gone.

To be clear, pls exclude it on platforms where it doesn't work on Semeru infrastructure, but do it consistently and pls document it properly.

JasonFengJ9 commented:
> But if it's not failing for Semeru on Linux, why would we exclude it there? If it is failing on Linux, the openj9 issue(s) should mention this. We don't need to duplicate what adoptium is doing.

So farm Semeru failure was only observed on JDK11 AIX, this PR added it to `ProblemList_openjdk11-openj9.txt`, updated `ProblemList_openjdk17-openj9.txt`, also removed `java/nio/channels/AsyncCloseAndInterrupt.java` exclude from JDK24+ since JDK21 doesn't have it. It can be added back if JDK24+ fails this test.

@pshipton this is ready for another look.
